app/controllers/api/v1/user_blocks_controller.rb

```diff
@@ -0,0 +1,50 @@
+class Api::V1::UserBlocksController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  
+  def get_blocked_users
+    id = request.headers['id'].to_i
+    #query block request where blocker_id is equal to user Id and blocked_id is equal to User id.
+    blockers = UserBlock.where(blocker_id: id)
+    blocked_bys = UserBlock.where(blocked_id: id)
+    
+    #declare an array for all the blocked users
+    user_array = []
+    #each through the blockers and push the user to the user_array
+    blockers.each do |block|
+        user = User.find_by(id: block[:blocked_id])
+        user = {id: user[:id], firstName: user[:first_name], lastName: user[:last_name]}
+        user_array.push(user)
+    end
+    blocked_bys.each do |block|
+      user = User.find_by(id: block[:blocker_id])
+      user = {id: user[:id], firstName: user[:first_name], lastName: user[:last_name]}
+      user_array.push(user)
+    end
+    render json: user_array.uniq
+  end
+  def get_blocked_by
+    id = request.headers['id'].to_i
+    #query block request where blocker_id or blocked_id is equal to user Id.
+    blocks = UserBlock.where(blocked_id: id)
+    #declare an array for all the blocked users
+    user_array = []
+    #each through the blocks and push the user to the user_array
+    blocks.each do |block|
+        user = User.find_by(id: block[:blocker_id])
+        user = {id: user[:id], firstName: user[:first_name], lastName: user[:last_name]}
+        user_array.push(user)
+      end
+    render json: user_array
+  end
+  def create
+    block = UserBlock.create(
+      blocker_id: params[:blocker_id],
+      blocked_id: params[:blocked_id]
+    )
+    render json: block.as_json
+  end
+  def destroy
+    block = UserBlock.find_by(id: params[:id])
+    block.destroy
```

Comment:
What happens when the block is not found? The app will return an error because you are calling destroy on nil. Adding something like `block.destroy if block.present?` would protect from that.

Reply:
Is this ok?
```
if block.exists?
  block.destroy
end
``` 
I read https://stackoverflow.com/a/43774445 and it was saying .exists? is quicker than .present? .

Reply:
Not necessarily. Try running `UserBlock.find_by(id: 1).exists?` in your rails console and let me know what you get.

Reply:
`exists?` can only be called on a collection of records. In our case, we are returning a single UserBlock record through the `find_by` method, which is why `present?` makes sense here.